cmd/sys-mgmt-executor/main.go

```diff
@@ -0,0 +1,189 @@
+	internal.SupportSchedulerServiceKey:     "Scheduler",
+}
+
+// usage will print out the flag options for the app.
+// This function is based on usage.go (in internal / pkg / usage)
+func HelpCallback() {
+	msg := fmt.Sprintf(usageStr, os.Args[0])
+	fmt.Printf("%s\n", msg)
+	os.Exit(0)
+}
+
+func main() {
+
+	start := time.Now()
+
+	flag.Usage = HelpCallback
+	flag.Parse()
+
+	logs.LoggingClient.Info(AppOpenMsg)
+
+	// Time it took to start service
+	logs.LoggingClient.Info("Application started in: " + time.Since(start).String())
+
+	var service = ""
+	var operation = ""
+
+	if len(os.Args) > 2 {
+		service = os.Args[1]
+		operation = os.Args[2]
+
+		err := ExecuteDockerCommands(service, operation)
+
+		if err != nil {
+			logs.LoggingClient.Error("error performing ", operation, " on service ", service, " with error: ", err.Error())
+		} else {
+			logs.LoggingClient.Info("success performing ", operation, " on service: ", service)
+		}
+	}
+}
+
+func findDockerContainerStatus(service string, status string) bool {
+
+	var (
+		cmdOut []byte
+		err    error
+	)
+	cmdName := "docker"
+	cmdArgs := []string{"ps"}
+	if cmdOut, err = exec.Command(cmdName, cmdArgs...).CombinedOutput(); err != nil {
+		logs.LoggingClient.Error("error running the docker command", "error message", err.Error())
+		os.Exit(1)
+	}
+
+	dockerOutput := string(cmdOut)
+
+	// Find whether the container to start has started.
+	for _, line := range strings.Split(strings.TrimSuffix(dockerOutput, "\n"), "\n") {
+		if strings.Contains(line, service) {
+
+			if status == "Up" {
+				if strings.Contains(line, "Up") {
+					logs.LoggingClient.Info("container started", "service name", service, "details", line)
+					return true
+				} else {
+					logs.LoggingClient.Warn("container NOT started", "service name", service)
+					return false
+				}
+			} else if status == "Exited" {
+				if strings.Contains(line, "Exited") {
+					logs.LoggingClient.Info("container stopped", "service name", service, "details", line)
+					return true
+				} else {
+					logs.LoggingClient.Warn("container NOT stopped", "service name", service)
+					return false
+				}
+			}
+		}
+	}
+	return false
+}
+
+func ExecuteDockerCommands(service string, operation string) error {
+	_, knownService := services[service]
+
+	if knownService {
+		err := runDockerCommands(service, services[service], operation)
+		logs.LoggingClient.Error(fmt.Sprintf("service %s ran into error while running Docker command: %v", service, err))
+		return err
+	} else {
+		err := errors.New("is an unknown service for which request was made to run Docker command")
+		logs.LoggingClient.Error(fmt.Sprintf("the service %s: %v", service, err))
+		return err
+	}
+}
+
+func runDockerCommands(service string, dockerService string, operation string) error {
+
+	var (
+		err    error
+		cmdDir string
+	)
+
+	cmdName := "docker"
+
+	cmdArgs := []string{operation, dockerService}
+	cmd := exec.Command(cmdName, cmdArgs...)
```

Comment:
I appreciate the simplicity of this refactoring without the big if/else statement, however we still need to perform verification of the operation that is being provided to us, as currently the operation is not verified at all. I consider this a security vulnerability because if there was some bug in the SMA that allowed specifying any command to this binary, they could effectively run any docker command they want. Performing validation here on what docker commands are executed is important to ensure that we only run `"docker start ...`", `"docker stop ..."`, etc.

More explanation can be found on this issue: https://github.com/edgexfoundry/edgex-go/issues/1129